File: python/quickstart/reconstruct_credential.py
#!/usr/bin/env python3
"""
Reconstruct a Claim Credential - Veritier Quickstart (Python)
=============================================================
Public GET needs no API key. JWKS lists current and previous Ed25519 kids.
Issuer revoke is authenticated and does not delete the workpaper.

Usage:
  python reconstruct_credential.py crc_...
  python reconstruct_credential.py crc_... --revoke   # needs VERITIER_API_KEY

Docs: https://veritier.ai/docs#attestation
"""

import os
import sys
import httpx
from dotenv import load_dotenv

load_dotenv()

API_URL = "https://api.veritier.ai"
API_KEY = os.getenv("VERITIER_API_KEY", "")

args = [a for a in sys.argv[1:] if a != "--revoke"]
do_revoke = "--revoke" in sys.argv[1:]
cred_id = args[0] if args else os.getenv("VERITIER_CREDENTIAL_ID", "")

if not cred_id:
    print("Usage: python reconstruct_credential.py crc_... [--revoke]")
    sys.exit(1)

jwk = httpx.get(f"{API_URL}/v1/credentials/.well-known/jwk", timeout=15.0)
if jwk.status_code != 200:
    print(f"JWKS error ({jwk.status_code}): {jwk.text}")
    sys.exit(1)
kids = [k.get("kid") for k in jwk.json().get("keys", [])]
print(f"JWKS kids: {kids}\n")

pub = httpx.get(f"{API_URL}/v1/credentials/{cred_id}", timeout=15.0)
if pub.status_code != 200:
    print(f"GET error ({pub.status_code}): {pub.text}")
    sys.exit(1)

data = pub.json()
print(f"id:              {data.get('id')}")
print(f"decision:        {data.get('decision')}")
print(f"procedure:       {data.get('procedure')}")
print(f"action_id:       {data.get('action_id')}")
print(f"engine_version:  {data.get('engine_version')}")
print(f"signature_valid: {data.get('signature_valid')}")
print(f"revoked:         {data.get('revoked')}")
print(f"as_of:           {data.get('as_of')}")
print(f"public page:     https://veritier.ai/c/{cred_id}")
print()
print("This is a workpaper, not a judicial finding.")

if data.get("revoked"):
    print("Issuer revoked this credential. Do not treat the decision as current.")

if do_revoke:
    if not API_KEY:
        print("\n--revoke requires VERITIER_API_KEY (same account that issued it).")
        sys.exit(1)
    revoked = httpx.post(
        f"{API_URL}/v1/credentials/{cred_id}/revoke",
        headers={"Authorization": f"Bearer {API_KEY}"},
        timeout=15.0,
    )
    print(f"\nrevoke HTTP {revoked.status_code}: {revoked.text[:300]}")

File: javascript/quickstart/reconstruct_credential.mjs
/**
 * Reconstruct a Claim Credential - Veritier Quickstart (JavaScript)
 * ==================================================================
 * Public GET needs no API key. JWKS lists current and previous Ed25519 kids.
 *
 * Usage:
 *   node reconstruct_credential.mjs crc_...
 *   node reconstruct_credential.mjs crc_... --revoke
 *
 * Docs: https://veritier.ai/docs#attestation
 */

import "dotenv/config";

const API_URL = "https://api.veritier.ai";
const API_KEY = process.env.VERITIER_API_KEY || "";
const args = process.argv.slice(2).filter((a) => a !== "--revoke");
const doRevoke = process.argv.includes("--revoke");
const credId = args[0] || process.env.VERITIER_CREDENTIAL_ID || "";

if (!credId) {
  console.error("Usage: node reconstruct_credential.mjs crc_... [--revoke]");
  process.exit(1);
}

const jwk = await fetch(`${API_URL}/v1/credentials/.well-known/jwk`);
if (!jwk.ok) {
  console.error(`JWKS error (${jwk.status}): ${await jwk.text()}`);
  process.exit(1);
}
const kids = ((await jwk.json()).keys || []).map((k) => k.kid);
console.log(`JWKS kids: ${kids.join(", ")}\n`);

const pub = await fetch(`${API_URL}/v1/credentials/${credId}`);
if (!pub.ok) {
  console.error(`GET error (${pub.status}): ${await pub.text()}`);
  process.exit(1);
}

const data = await pub.json();
console.log(`id:              ${data.id}`);
console.log(`decision:        ${data.decision}`);
console.log(`procedure:       ${data.procedure}`);
console.log(`action_id:       ${data.action_id}`);
console.log(`engine_version:  ${data.engine_version}`);
console.log(`signature_valid: ${data.signature_valid}`);
console.log(`revoked:         ${data.revoked}`);
console.log(`as_of:           ${data.as_of}`);
console.log(`public page:     https://veritier.ai/c/${credId}`);
console.log("\nThis is a workpaper, not a judicial finding.");

if (data.revoked) {
  console.log("Issuer revoked this credential. Do not treat the decision as current.");
}

if (doRevoke) {
  if (!API_KEY) {
    console.error("\n--revoke requires VERITIER_API_KEY (same account that issued it).");
    process.exit(1);
  }
  const revoked = await fetch(`${API_URL}/v1/credentials/${credId}/revoke`, {
    method: "POST",
    headers: { Authorization: `Bearer ${API_KEY}` },
  });
  console.log(`\nrevoke HTTP ${revoked.status}: ${(await revoked.text()).slice(0, 300)}`);
}
